technologydata/utils/currencies.py

```diff
@@ -0,0 +1,554 @@
+            If the specified deflator function name is not found in the deflation function registry.
+
+        Examples
+        --------
+        >>> data = pd.DataFrame({
+        ...     'unit': ['WB_USD_2020', 'EUR_2021', 'JPY_2019'],
+        ...     'value': [100, 200, 300],
+        ...     'region': ['USA', 'FRA', 'JPN']
+        ... })
+        >>> adjusted_data = Currencies.adjust_currency(
+        ...     base_year_val=2022,
+        ...     target_currency='USD',
+        ...     data=data,
+        ...     deflator_function_name='International Monetary Fund',
+        ... )
+        >>> adjusted_data['unit']
+        0    USD_2022
+        1    USD_2022
+        2    USD_2022
+        Name: unit, dtype: object
+
+        """
+        # Specify the path where deflator and exchange data will be saved
+        if pydeflate_path is not None:
+            pyd.set_pydeflate_path(pydeflate_path)
+            pydeflate_path.mkdir(parents=True, exist_ok=True)
+        else:
+            raise ValueError(
+                "The path where the deflator and exchange data will be saved is None"
+            )
+
+        # Validate columns presence
+        required_columns = {"unit", "value", "region"}
+        if not required_columns.issubset(data.columns):
+            missing = required_columns - set(data.columns)
+            raise ValueError(f"Input dataFrame is missing required columns: {missing}")
+
+        # Create a copy of the original data to avoid modifying input directly
+        results = data.copy()
+
+        # Select rows that correspond to a unit column that fulfills the format <3-letter currency code>-<currency year>
+        has_currency_mask = (
+            results["unit"].apply(Currencies.extract_currency_unit).notna()
+        )
+        currency_rows = results.loc[has_currency_mask].copy()
+
+        if currency_rows.empty:
+            logger.warning("No rows contain a valid currency unit.")
+        else:
+            # For each row, extract currency year and currency from the unit column
+            currency_rows[["currency_code", "currency_year"]] = (
+                currency_rows["unit"]
+                .apply(Currencies.extract_currency_unit)
+                .str.split(separator, expand=True)
+            )
+
+            # Cast 'currency_year' column to integer type
+            currency_rows["currency_year"] = pd.to_numeric(
+                currency_rows["currency_year"], errors="coerce"
+            ).astype(int)
+
+        target_country = Currencies.get_country_from_currency(target_currency)[0]
+        currency_rows["source_country"] = currency_rows["currency_code"].apply(
+            lambda c: Currencies.get_country_from_currency(c)[0]
+            if Currencies.get_country_from_currency(c)
+            else None
+        )
+
+        deflate_row_func = Currencies.get_deflate_row_function(
+            base_year=base_year_val,
+            deflator_name=deflator_function_name.casefold(),
```

Comment:
```suggestion
            deflator_name=deflator_function_name,
```

See note above - I moved it as close to the actual comparison as possible